elastic/apm-server
# Iterate package

## Motivation/summary

This does a few things:

- Splits metrics data stream into app metrics, internal and profiles. This is following the indexing strategy proposal.
- Adds the dataset property to the manifest files so the right templates are created.
- Sort and dedup fields (within each yml file). Fields are now flattened. 
- Generates data stream pipelines out of our `definitions.json` file

~**Missing from this PR / WIP**: Pipelines are tied to data streams, we need to copy the ones that are used for other event types to the right data stream folder. Right now all are under `traces`. This shouldn't block the review.~


## Checklist


I have considered changes for:

## How to test these changes

In short: run the package, install it, and check that all templates and pipelines installed are correct.

## Related issues

Part of #4004
